From 12234ff855688281369b06229541bc7d5155a1d1 Mon Sep 17 00:00:00 2001
From: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
Date: Mon, 29 Jul 2019 16:10:59 +0100
Subject: [PATCH 30/34] imx6: imx6ul_picopi: atf_uboot_compat Add simple
 compatibility header

In order to import u-boot DDR initialization code we will need to provide
some very simple is_imx6ul() type functions.

We can inline those simple functions in a compatibility header licensed BSD
like the rest of ATF.

Signed-off-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
---
 .../imx6/imx6ul_picopi/include/atf_uboot_compat.h  | 23 ++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100644 plat/imx/imx6/imx6ul_picopi/include/atf_uboot_compat.h

diff --git a/plat/imx/imx6/imx6ul_picopi/include/atf_uboot_compat.h b/plat/imx/imx6/imx6ul_picopi/include/atf_uboot_compat.h
new file mode 100644
index 0000000..2018701
--- /dev/null
+++ b/plat/imx/imx6/imx6ul_picopi/include/atf_uboot_compat.h
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2018, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef __ATF_UBOOT_COMPAT_H__
+#define __ATF_UBOOT_COMPAT_H__
+
+#define is_mx6sx() false
+#define is_mx6ul() true
+#define is_mx6ull() false
+#define is_mx6sl() false
+#define is_mx6dq() false
+#define is_mx6dqp() false
+
+#define readl(x) mmio_read_32((uintptr_t)x)
+
+#define DDR_CLOCK_MHZ 400000000
+#define mxc_get_clock(x) DDR_CLOCK_MHZ
+
+#endif /* __ATF_UBOOT_COMPAT_H__ */
+
-- 
2.7.4

